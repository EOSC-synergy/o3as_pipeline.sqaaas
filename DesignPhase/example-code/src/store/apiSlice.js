// Import the RTK Query methods from the React-specific entry point
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'

// Define our single API slice object
export const apiSlice = createApi({
  // The cache reducer expects to be added at `state.api` (already default - this is optional)
  reducerPath: 'api',
  
  baseQuery: fetchBaseQuery({ baseUrl: 'https://api.o3as.fedcloud.eu/api/v1' }),
  
  // The "endpoints" represent operations and requests for this server
  endpoints: builder => ({
    
    getModels: builder.query({
      query: () => '/models'
    }),
    getPlots: builder.query({
      query: () => '/plots'
    }),
    getRawPlotData: builder.mutation({
      query: payload => ({
        // /data/tco3_return?begin=1959&end=2100&lat_min=-90&lat_max=90
        url: `/data/${payload.plotType}?begin=1959&end=2100&lat_min=${payload.lat_min}&lat_max=${payload.lat_max}`,
        method: "POST",
        body: payload.modelList,
      })
    }),
    
    getBuiltPlotData: builder.mutation({
      query: payload => ({
        
        url: `/plots/${payload.plotType}?begin=1959&end=2100&lat_min=${payload.lat_min}&lat_max=${payload.lat_max}`,
        method: "POST",
        body: payload.modelList,
      })
    }),
    
  })
})

// Export the auto-generated hook for the query endpoints
export const { 
  useGetPostsQuery,
  useGetPostQuery,
  useGetBuiltPlotDataMutation,
  useGetRawPlotDataMutation,
 } = apiSlice