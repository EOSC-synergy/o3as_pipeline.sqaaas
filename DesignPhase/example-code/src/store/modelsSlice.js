import { createSlice } from "@reduxjs/toolkit"

const initialState = {
    "octs": { // plot name (id)
        "all": { // model group
            name: "Somehting",
            modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
            models: { // models is lookup table
                "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                    institute: "IMK",
                    dataset: {
                        x: [42],
                        y: [42],
                    },
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                    color: "#000000",
                    plotStyle: "solid",
                }
            },
            hidden: false,
            derivativeVisible: false,
            meanVisible: false,
            medianVisible: false,
            percentileVisible: false
        }
    }
}

const modelsSlice = createSlice({
    name: "models",
    initialState,
    reducers: {
        updatedModelGroup(state, action) { },
        addedModelGroup(state, action) { },
        setVisibility(state, action) { },
        setStatisticalValuesIncluded(state, action) { },
        addStatisticalValue(state, action) { },
        removeStatisticalValue(state, action) { },
    }
})

export const { 
    updatedModelGroup,
    addedModelGroup,
    setVisibility, 
    setStatisticalValuesIncluded, 
    addStatisticalValue, 
    removeStatisticalValue 
} = modelsSlice.actions

export default modelsSlice.reducer

export const selectCurrentModelGroups = state => state.models[state.plot.plotId]
export const selectCurrentModelGroup = (state, groupId) => state.models[state.plot.plotId][groupId]