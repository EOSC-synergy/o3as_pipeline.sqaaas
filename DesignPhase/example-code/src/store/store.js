import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice'
import modelsReducer from './modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})