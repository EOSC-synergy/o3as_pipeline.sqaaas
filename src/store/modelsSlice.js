import { createSlice } from "@reduxjs/toolkit";

const STATISTICAL_VALUES = ["mean", "median", "derivative", "percentile"];

const initialState = {
    // currently active plot
    plotId: "tco3_zm",
    settings: {
        "tco3_zm": { 
            // this objects holds key-value-pairs, the keys being the model-group 
            // identifier and the values being the settings for each group 
            "all": { 
                // model group storing all information until it is possible 
                // to implement more model groups
                name: "All OCTS models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        color: "#000000",
                        plotStyle: "solid",
                        isVisible: true, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        },
        "tco3_return": {
            "all": { 
                name: "All Return/Recovery models",
                modelList: ["CCMI-1_ACCESS_ACCESS-CCM-refC2"],
                models: { // models is lookup table
                    "CCMI-1_ACCESS_ACCESS-CCM-refC2": { // single model
                        institute: "IMK",
                        dataset: {
                            x: [42],
                            y: [42],
                        },
                        hidden: false, // show/hide individual models from a group
                        mean: true,
                        derivative: true,
                        median: true,
                        percentile: true,
                        color: "#000000",
                        plotStyle: "solid",
                    }
                },
                hidden: false, // show/hide complete group
                visibileSV: { // lookup table so the reducer impl. can be more convenient
                    mean: true,
                    derivative: true,
                    median: true,
                    percentile: true,
                }
            }
        }
    }
}

const modelsSlice = createSlice({
    name: "models",
    initialState,
    reducers: {
        // set models
        // add models

        addModels(state, action) { 
            const { groupId, newModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            // Add model to modelList if it is not already included
            for (var i = 0; i < newModelList.length; i++) {
                if (activeModelList.indexOf(newModelList[i]) !== -1) {
                    continue;
                }

                activeModelList.push(newModelList[i]);
            }

            // ---------------------------------------------------------
            // TODO: Add Model information to the model lookup table
            // ---------------------------------------------------------


        }, // expects groupID, list of models 
        removeModels(state, action) { 
            const { groupId, removeModelList }  = action.payload;
            const activeSettings = state.settings[state.plotId];
            const selectedGroup = activeSettings[groupId];
            const activeModelList = selectedGroup.modelList;

            if (selectedGroup === undefined) {
                throw `Tried to access model-group with groupID "${groupId}" that is not yet defined!`;
            }

            for (var i = 0; i < removeModelList.length; i++) {
                // Remove model from modelList
                activeModelList = activeModelList.filter(item => item !== removeModelList[i]);

                // Delete model object from models
                delete selectedGroup.models[removeModelList[i]];
            }

        }, // expects groupdID, list of models
        updatedModelGroup(state, action) { }, // implementing this requires more knowledge about the UI
        addedModelGroup(state, action) { },
        setVisibility(state, action) { 
            const { groupID, modelID, isVisible } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            } 
            activeModel.isVisible = isVisible;
        },
        setStatisticalValuesIncluded(state, action) { // this is for an individual model
            // svType has to be a string of the form: 
            // mean | derivative | median | percentile
            const { groupID, modelID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            const activeModel = activeSettings[groupID].models[modelID];
            if (activeModel === undefined) {
                throw `tried to access model with modelID "${modelID}" that is not yet defined!`;
            }
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeModel[svType] = isIncluded;
        },
        setStatisticalValueForGroup(state, action) { // this is for an entire group
            const { groupID, svType, isIncluded } = action.payload;
            const activeSettings = state.settings[state.plotId];
            if (STATISTICAL_VALUES.includes(svType)) { // svType doesn't represent a valid statistical value
                throw `tried to set statistial value "${svType}" that is not a valid statistical value (${STATISTICAL_VALUES.join("|")})`;
            }
            activeSettings[groupID].visibileSV[svType] = isIncluded;
        },
    }
})

export const { 
    addModels,
    removeModels,
    updatedModelGroup,
    addedModelGroup,
    setVisibility, 
    setStatisticalValuesIncluded, 
    setStatisticalValueForGroup
} = modelsSlice.actions;

export default modelsSlice.reducer;

export const selectCurrentModelGroups = state => state.models[state.plot.plotId];
export const selectCurrentModelGroup = (state, groupId) => state.models[state.plot.plotId][groupId];