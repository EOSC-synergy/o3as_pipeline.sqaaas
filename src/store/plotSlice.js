import { createSlice } from "@reduxjs/toolkit"

/**
 * The initial state of the plotSlice defines the data structure in the 
 * store. Each plot has its own settings i.e. title, location etc.
 */
const initialState = {
    
    plotId: "tco3_zm", // currently active plot
    // maps plotids to their settings
    settings: {
        "tco3_zm": {
            name: "OCTS", // should show up in the drop down menu
            title: "OCTS Plot", // the title shown in the apexcharts generated chart
            location: {
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 1960,
                maxX: 2100,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        },
        "tco3_return": {
            name: "Return/Recovery",
            title: "Return/Recovery Plot",
            location: { // custom user defined region
                minLat: -90,
                maxLat: 90
            },
            displayXRange: {
                minX: 0,
                maxX: 0,
            },
            displayYRange: {
                minY: 200,
                maxY: 400,
            },
            months: [
                1
            ],
        }
    }
}

/**
 * The plotSlice is generated by the redux toolkit. The reducers are defined here
 * and the corresponding actions are auto-generated.
 */
const plotSlice = createSlice({
    name: "plot",
    initialState,
    reducers: {
        setActivePlotId(state, action) { // e.g. dispatch(setActivePlotId({id: "tco3_zm"}))
            const { id } = action.payload
            state.plotId = id
        }, 
        setTitle(state, action) { // e.g. dispatch(setActivePlotId({title: "OCTS Plot"}))
            const { title } = action.payload
            state.settings[state.plotId].title = title
        },
        setLocation(state, action) { 
            const {minLat, maxLat} = action.payload
            const location = state.settings[state.plotId].location
            location.minLat = minLat
            location.maxLat = maxLat
        },
        setDisplayXRange(state, action) { 
            const {minX, maxX} = action.payload
            const displayXRange = state.settings[state.plotId].displayXRange
            displayXRange.minX = minX
            displayXRange.maxX = maxX
        },
        setDisplayYRange(state, action) { 
            const {minY, maxY} = action.payload
            const displayYRange = state.settings[state.plotId].displayYRange
            displayYRange.minY = minY
            displayYRange.maxY = maxY
        },
        setMonths(state, action) { 
            state.settings[state.plotId].months = action.payload.months
        },
    }   
})


/**
 * The here listed actions are exported and serve as an interface for
 * the view (our react components).
 */
export const {
    setActivePlotId,
    setTitle,
    setLocation,
    setDisplayXRange,
    setDisplayYRange,
    setMonths,
} = plotSlice.actions

/**
 * The reducer combining all reducers defined in the plot slice. 
 * This has to be included in the redux store, otherwise dispatching 
 * the above defined actions wouldn't trigger state updates.
 */
export default plotSlice.reducer

/**
 * This selectors allows components to select the current plot id
 * from the store. The plot id is a string using the same naming as the
 * o3as api e.g. tco3_zm or tco3_return
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot id
 */
export const selectPlotId = state => state.plot.plotId

/**
 * This selectors allows components to select the current plot name
 * from the store. The plot name is the humand readable version specifying 
 * the plot e.g. OCTS or Return/Recovery
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot name
 */
export const selectPlotName = state => state.plot.settings[state.plot.plotId].name

/**
 * This selectors allows components to select the current plot title
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {string} the current active plot title
 */
export const selectPlotTitle = state => state.plot.settings[state.plot.plotId].title

/**
 * This selectors allows components to select the current plot location
 * from the store. The location is an object containg a minLat and maxLat attribute.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current location that includes a minLat and maxLat attribute.  
 */
export const selectPlotLocation = state => state.plot.settings[state.plot.plotId].location

/**
 * This selectors allows components to select the current x range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minX and maxX
 */
export const selectPlotXRange = state => state.plot.settings[state.plot.plotId].displayXRange

/**
 * This selectors allows components to select the current y range
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {object} holds the current x range that includes minY and maxY
 */

export const selectPlotYRange = state => state.plot.settings[state.plot.plotId].displayYRange

/**
 * This selectors allows components to select the current selected months
 * from the store.
 * 
 * @param {object} state the global redux state
 * @returns {array} array of strings describing the current selected months
 */

export const selectPlotMonths = state => state.plot.settings[state.plot.plotId].months
