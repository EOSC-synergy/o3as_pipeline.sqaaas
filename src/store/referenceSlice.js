import { createSlice } from "@reduxjs/toolkit"

const initialState = {
    "OCTS": {
        year: 1980,
        model: "CCMI-1_ACCESS_ACCESS-CCM-refC2",
        visibile: false,
        offsetApplied: false,
    }
}

const referenceSlice = createSlice({
    name: "reference",
    initialState,
    reducers: {
        setReferenceYear(state, action) {
            const {year} = action.payload;
            state.OCTS.year = year;
         },
        setReferenceModel(state, action) { 
            const {model} = action.payload;
            state.OCTS.model = model;
        },
        setVisibile(state, action) { },
        setOffsetApplied(state, action) { },
    }
})

export const {
    setReferenceYear,
    setReferenceModel,
    setVisibile,
    setOffsetApplied,
} = referenceSlice.actions

export default referenceSlice.reducer

export const selectCurrentReferenceSettings = state=> state.reference[state.plot.plotId]
