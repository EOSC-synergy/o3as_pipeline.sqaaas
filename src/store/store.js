import { configureStore } from '@reduxjs/toolkit'
import plotReducer from './plotSlice/plotSlice'
import modelsReducer from './modelsSlice/modelsSlice'
import referenceReducer from './referenceSlice'


export default configureStore({
  reducer: {
    plot: plotReducer,
    models: modelsReducer,
    reference: referenceReducer,
  }
})