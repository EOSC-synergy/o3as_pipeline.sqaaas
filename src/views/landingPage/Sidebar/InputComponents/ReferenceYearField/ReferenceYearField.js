import React from "react";
import { Grid, Typography, FormControl, TextField, Checkbox } from "@mui/material";
import { useDispatch } from "react-redux"
import { useSelector } from "react-redux";
import { setYear, setVisibility, selectVisibility } from "../../../../../store/referenceSlice/referenceSlice";
import { END_YEAR, START_YEAR } from "../../../../../utils/constants";
import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
import MuiVisibilityIcon from '@mui/icons-material/Visibility';

/**
 * Enables the user to select a reference year.
 * @param {Object} props
 * @param {function} props.reportError - function to handle errors
 * @returns {JSX.Element} a jsx containing a text field to select the reference year
 */
function ReferenceYearField(props) {
    /**
     * A dispatch function to dispatch actions to the redux store.
     */
    const dispatch = useDispatch();

    /**
     * The selected reference year from the redux store.
     */
    const selectedYear = useSelector(state => state.reference.settings.year);

    /**
     * The selected visibility of the reference line from the redux store
     */
    const refYearVisible = useSelector(selectVisibility)

    /**
     * Handles the change of the reference year field if it is modified.
     */
    const handleChangeForRefYear = (event) => {
        dispatch(setYear({year: event.target.value}));
    };

    /**
     * Handles the change of the reference line visibility field if it is modified.
     */
    const handleShowRefLineClicked = (event) => {
        dispatch(setVisibility({visible: event.target.checked}))
    }

    return (
        <>
        <Grid container sx={{width: "90%", marginLeft: "auto", marginRight: "auto", marginTop: "5%"}}>
            <Grid item xs={5}>
            <Typography>Reference year:</Typography>
            </Grid>
            <Grid item xs={7} sx={{mt: "-8px"}}>
                <FormControl sx={{width: '35%'}}>
                    <TextField
                        variant="outlined"
                        id="outlined-basic"
                        size="small"
                        value={selectedYear}
                        onChange={handleChangeForRefYear}
                        error={selectedYear < START_YEAR || selectedYear > END_YEAR}
                        helperText={selectedYear < START_YEAR ? `<${START_YEAR}` : (selectedYear > END_YEAR ? `>${END_YEAR}` : '')}
                    />
                </FormControl>

                <FormControl>
                    <Checkbox
                        icon={<VisibilityOffIcon data-testid="RefLineInvisibleCheckbox" />}
                        checkedIcon={<MuiVisibilityIcon />}
                        onClick={handleShowRefLineClicked} 
                        //checked={refYearVisible} 
                    />
                </FormControl>
                
            </Grid>
        </Grid>
        </>
    );
}

export default ReferenceYearField;
