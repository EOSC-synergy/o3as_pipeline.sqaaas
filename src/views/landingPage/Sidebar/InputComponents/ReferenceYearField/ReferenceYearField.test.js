import React from 'react';
import ReferenceYearField from './ReferenceYearField';
import { render } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import { createTestStore } from '../../../../../store/store';
import { Provider } from "react-redux";

let store;

beforeEach(() => {
  store = createTestStore();
});

it('Component renders without crashing', () => {
    render(<Provider store={store}><ReferenceYearField /></Provider>);

});


// Snapshot test
it('renders correctly', () => {
    const { container } = render(<Provider store={store}><ReferenceYearField /></Provider>);
    expect(container).toMatchSnapshot();
  });



test.todo('changes the reference year correctly');