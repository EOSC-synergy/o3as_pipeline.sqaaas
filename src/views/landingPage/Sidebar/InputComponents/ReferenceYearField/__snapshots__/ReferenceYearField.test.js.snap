// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-wts08w-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Reference year:
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7 css-a0yaxr-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root css-1st3y2y-MuiFormControl-root"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              id="outlined-basic"
              type="text"
              value="1980"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-nnbavb"
              >
                <span
                  class="notranslate"
                >
                  â€‹
                </span>
              </legend>
            </fieldset>
          </div>
          
        </div>
      </div>
      <div
        class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
      >
        <span
          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="RefLineInvisibleCheckbox"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
